pkg/encryption/blockcipher/blockcipher_aes_ctr.go

```diff
@@ -129,13 +129,15 @@ func (bc *AESCTRLayerBlockCipher) init(encrypt bool, reader io.Reader, opts Laye
 	bc.stream = cipher.NewCTR(block, nonce)
 	bc.err = nil
 	bc.hmac = hmac.New(sha256.New, key)
-	bc.setHmac = setHmac
-	bc.expHmac = expHmac
+	bc.expHmac, _ = opts.GetOpt("hmac")
+	bc.doneEncrypting = false
 
 	lbco := LayerBlockCipherOptions{
-		SymmetricKey: key,
-		CipherOptions: map[string][]byte{
-			"nonce": nonce,
+		Private: PrivateLayerBlockCipherOptions{
```

Comment:
Aren't we only storing private data here anyway, since we are encrypting this? Isn't the 'Private' here redundant?

Reply:
Ah, I see lbco.PrivateData will be JSON marshalled and encrypted.